### PR TITLE
Feature/dropdown search

### PR DESCRIPTION
**Pull request by sixskulls**
Adding search to displayed property as well as possibility to search custom properties for angular dropdown component, will need added implementation for React side.
image

✅ Closes: #473

### CODE REVIEW

**Comment by sebopensource** [comment]
👋 Thanks for creating a pull request! 🚀 Checkout the storybooks we've created for it: <ul><li><a target="_blank" href="https://sebgroup.github.io/green/pr-573/chlorophyll/">chlorophyll</a></li><li><a target="_blank" href="https://sebgroup.github.io/green/pr-573/react/">react</a></li><li><a target="_blank" href="https://sebgroup.github.io/green/pr-573/angular/">angular</a></li></ul>

File: libs/extract/src/lib/dropdown/reducers.ts [hunk 1]
```diff
@@ -318,3 +325,25 @@ export const validate = (dropdown: AbstractDropdown, validator: any) => reduce(
...
   }
 )
+
+export const search = (dropdown: AbstractDropdown, searchInput: string ) : AbstractDropdown => reduce(
+  dropdown, {
+    options: dropdown.options.map(option => {
+      let isMatch = false;
+      const propNames = [...[dropdown.display], ...dropdown.searchableProperties||[]]
+
+      propNames.every(prop=> {
```
Comment (Maximilian-R) [inline libs/extract/src/lib/dropdown/reducers.ts:335]: The inverted isMatch logic in the every function is a bit hard to follow along. I would suggest to replace `every` with `some` and put the empty searchinput condition separately. Something like this:
```
    const isMatch =
    searchInput.length === 0 ||
    propNames.some((prop) =>
      option[prop]?.toString().toLowerCase().includes(searchInput.toLowerCase())
    );

**Review by Maximilian-R — COMMENTED** [review]
Looks really nice! :) I added a few comments that are not required to fix but simplifies the code a little

**Review by Maximilian-R — APPROVED** [review]
